File: c++/SORTING/selection_sort.cpp
// SORTING OF ARRAY
#include<iostream>
using namespace std; 
void sortElement(int arr[], int size)
{
    int minIndex,i;
    for(int i=0; i<size-1; i++)
    {
        minIndex=i;
        for(int j=i+1; j<size; j++)
        {
            if(arr[j]<arr[minIndex])
                minIndex = j;
        }
        swap(arr[minIndex],arr[i]);
    }
    cout<<"Array after sorting: ";
    for(int i=0; i<size; i++)
    {
        cout<<arr[i]<<" ";
    }
}
int main()
{
    int key;
    int arr[6]={1,109,45,60,23,9};
    
    sortElement(arr, 6);
}

File: javaPractice/sorting/SelectionSort.java
package sorting;

public class SelectionSort {
    
    public static void main(String[] args) {
        int[] arr = {23, 171, 87, 19, 129, 786, 26, 468};
        sort(arr);
    }
    
    public static void sort(int[] arr) {
        int temp;
        for (int i = 0; i < arr.length; i++) {
            int minIndex = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[minIndex])
                    minIndex = j;
            }
            temp = arr[minIndex];
            arr[minIndex] = arr[i];
            arr[i] = temp;
        }
        for (int i : arr)
            System.out.print(i + " ");
    }

}
